common/src/main/java/com/wynntils/models/lootrun/LootrunModel.java

```diff
@@ -135,39 +175,78 @@ public void onChatMessage(ChatMessageReceivedEvent event) {
 
     @SubscribeEvent
     public void onWorldStateChanged(WorldStateEvent event) {
+        // The world state event is sometimes late compared to lootrun events (beacons, scoreboard)
+        // Reseting once when leaving the class is enough
+        if (event.getNewState() == WorldState.WORLD) return;
+
         lootrunCompletedBuilder = null;
         lootrunFailedBuilder = null;
 
         lootrunningState = LootrunningState.NOT_RUNNING;
-        location = null;
         taskType = null;
+        beaconUpdates = new HashMap<>();
+        beacons = new HashMap<>();
+        LOOTRUN_BEACON_COMPASS_PROVIDER.reloadTaskMarkers();
 
         selectedBeacons = new TreeMap<>();
     }
 
+    @SubscribeEvent
+    public void onBeaconMoved(BeaconEvent.Moved event) {
```

Comment:
I changed the Move event to have both the old and the new beacon. I did not realize that one of them were still accessible using the "anonymous" `getBeacon` method from the parent.

Are you trying to access the old or the new beacon here?

Reply:
The new one.. that's partially why it's been so bad.. either way the new PR has accurate predictions but I will update this.